### PR TITLE
fix. typos in _protocol , _object_tracker, _object_records , and  _retaining_path.dart

### PR DESCRIPTION
**Pull request by droidbg**
### Description
This PR fixes typo in 
- `_protocol.dart`.
- `_object_tracker.dart`.
- `_object_records.dart`.
- `_retaining_path.dart`

## What type of PR is this?

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

---

- [X] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by polina-c — APPROVED** [review]
Thank you!!!

**Comment by polina-c** [comment]
The tests will stop failing after merge of this: https://github.com/dart-lang/leak_tracker/pull/165

**Comment by polina-c** [comment]
@droidbg , would you mind to fix conflicts?

File: pkgs/leak_tracker/lib/src/leak_tracking/_primitives/_retaining_path/_retaining_path.dart [hunk 1]
```diff
@@ -7,7 +7,7 @@ import 'dart:developer';
 import 'dart:isolate';
 import 'package:vm_service/vm_service.dart' hide Isolate;
 
-/// Returns retaining path for an object, if it can be detected.
+/// Returns retainig path for an object, if it can be detected.
```
Comment (polina-c) [inline pkgs/leak_tracker/lib/src/leak_tracking/_primitives/_retaining_path/_retaining_path.dart]: `retainig` does not seem to be a word.
I googled it: https://www.google.com/search?q=retainig

Comment (droidbg) [reply pkgs/leak_tracker/lib/src/leak_tracking/_primitives/_retaining_path/_retaining_path.dart]: yes `retainig` was originally present in code, to resolve merge conflict I revert that and after again pushed the correct one.

**Review by polina-c — APPROVED** [review]

**Pull request merged**